src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -229,6 +229,15 @@ public Packaging.PackageBuilder CreateBuilder(string basePath, NuGetVersion vers
 
             var projectAuthor = InitializeProperties(builder);
 
+            // Set version based on version argument from console?
+            if (version != null)
+            {
+                // make sure the $version$ placeholder gets populated correctly
+                _properties["version"] = version.ToString();
```

Comment:
ToFullString() is probably the best option here since it won't make any decisions on how this is to be used. It is ToString() + build metadata for semver 2.0.0 (you have to opt in to getting that extra data from NuGetVersion)